ghost/core/test/e2e-frontend/site_id_middleware.test.js

```diff
@@ -0,0 +1,90 @@
+const sinon = require('sinon');
+const supertest = require('supertest');
+const testUtils = require('../utils');
+const configUtils = require('../utils/configUtils');
+
+describe('Site id middleware execution', function () {
+    let request;
+
+    before(async function () {
+        configUtils.set('hostSettings:siteId', '123123');
+
+        // Ensure we do a forced start so that spy is in place when the server starts
+        await testUtils.startGhost({forceStart: true});
+
+        request = supertest.agent(configUtils.config.get('url'));
+    });
+
+    after(async function () {
+        sinon.restore();
+
+        configUtils.restore();
+
+        await testUtils.stopGhost();
+    });
+
+    describe('Using site-id middleware', function () {
```

Comment:
Should we add an extra safety check test for when `hostSettings:siteId` is explicitly not set (middleware should not be executed)?